Arrowgene.Ddon.GameServer/Handler/CraftStartEquipGradeUpHandler.cs

```diff
@@ -0,0 +1,273 @@
+#nullable enable
+using System.Linq;
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Logging;
+using System;
+using System.Collections.Generic;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class CraftStartEquipGradeUpHandler : GameRequestPacketHandler<C2SCraftStartEquipGradeUpReq, S2CCraftStartEquipGradeUpRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(CraftStartEquipGradeUpHandler));
+        private static readonly List<StorageType> StorageEquipNBox = new List<StorageType> {
+            StorageType.ItemBagEquipment, StorageType.StorageBoxNormal, StorageType.StorageBoxExpansion, StorageType.StorageChest, StorageType.CharacterEquipment, StorageType.PawnEquipment
+        };
+
+        private readonly ItemManager _itemManager;
+        private readonly Random _random;
+
+        public CraftStartEquipGradeUpHandler(DdonGameServer server) : base(server)
+        {
+            _itemManager = Server.ItemManager;
+            _random = Random.Shared;
+        }
+
+        public override S2CCraftStartEquipGradeUpRes Handle(GameClient client, C2SCraftStartEquipGradeUpReq request)
+
+        {
+            #region Initializing vars
+            Character character = client.Character;
+            string equipItemUID = request.EquipItemUID;
+            Item equipItem = Server.Database.SelectStorageItemByUId(equipItemUID);
+            uint charid = client.Character.CharacterId;
+            uint pawnid = request.CraftMainPawnID;
+
+            CDataMDataCraftGradeupRecipe json_data = Server.AssetRepository.CraftingGradeUpRecipesAsset
+                .SelectMany(recipes => recipes.RecipeList)
+                .Where(recipe => recipe.ItemID == equipItem.ItemId)
+                .First();
+
+            uint gearupgradeID = json_data.GradeupItemID;
+            uint goldRequired = json_data.Cost;
+            uint EquipRank = json_data.Unk0;
+            bool canContinue = true;
+            bool IsGreatSuccess = _random.Next(5) == 0; // 1 in 5 chance to be true, someone said it was 20%.
```

Comment:
Instead of having `_random` switch to using `Random.Shared.Next(5)`.

Reply:
_random is already set to Random.Shared, but yeah, you don't need to store it